src/libstd/sync/mutex.rs

```diff
@@ -450,9 +450,13 @@ impl<'a, T: ?Sized> Drop for MutexGuard<'a, T> {
 #[stable(feature = "std_debug", since = "1.16.0")]
 impl<'a, T: ?Sized + fmt::Debug> fmt::Debug for MutexGuard<'a, T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.debug_struct("MutexGuard")
-            .field("lock", &self.__lock)
-            .finish()
+        struct MutexFmt<'a, T: ?Sized>(&'a MutexGuard<'a, T>);
+        impl<'a, T: ?Sized + fmt::Debug> fmt::Debug for MutexFmt<'a, T> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.debug_struct("Mutex").field("data", &&*self.0).finish()
+            }
+        }
+        f.debug_struct("MutexGuard").field("lock", &MutexFmt(self)).finish()
```

Comment:
MutexGuard is basically a smart pointer, and Box/Arc/etc just delegate directly to the inner value and don't apply any extra layer at all. I'd probably just have this look like `fmt::Debug::fmt(&**self, f)`.

Reply:
I agree. :) Makes both the implementation and the output more readable. Just didn't want to change the output format, as I wasn't sure if people cared about it.